### ISSUE TITLE
Trying to redirect back once my form has been submitted / handled, but the old data is getting displayed.

### ISSUE THREAD

**Issue by Drillan767** [inertiajs/inertia-laravel#165]
Hello, I'm working on a form, and I want my user to be redirect back to this form once it's submitted.

Here is my (simplified) code from Vue:

(Note, there are both images and texts in this form)

```
<template>
  <form @submit.prevent="submit">
    <input type="text" v-model="landing.value1" />
    <input type="text" v-model="landing.value2" />
    <input type="submit" value="submit" />
  </form>
</template
<script>
export default {
  data () {
    landing: this.$inertia.form({
      ...this.$page.landing
    })
  },

  methods: {
    submit () {
      this.landing.post('/some/url');
      window.scrollTo({top: 0, behavior: 'smooth'});
    }
  }
}

</script>
```

Here is my Laravel controller:

```php
public function update (LandingRequest $request)
{
  $landing = Landing::first();
  $landing->value1 = $request->get('value1');
  $landing->value2 = $request->get('value2');
  $landing->save();

  return Redirect::route('landing.edit')->with('success', 'Landing correctly updated');
  // Also tried with (return Redirect::back()
}
```

When I submit my form, I get no error whatsoever and get back to the top of the form with my confirmation message displayed. However, in the mean time, my form has changed for its OLD value, and I need to refresh the page to get the new value.

I have the feeling that I'm missing something, yet, I tried some tweak without success, and can't even understand why or when the old value is getting loaded.

**Comment by reinink** [comment]
So, what you're doing here should totally work. It looks like you're using Jetstream, so there is some of the Inertia details abstracted away. Regardless, when you submit the form using `this.landing.post()`, you're making an Inertia request. When you redirect server-side back to the the edit page, the edit page should get the updated information about the `Landing` from the database, and pass those props back to the edit page (causing them to update).

The part I'm confused about is this:

> my form has changed for its OLD value

This shouldn't be the case. Are you, by any chance, using the `resetOnSuccess: true` feature of Jetstream?

**Comment by Drillan767** [comment]
Hello,

I am indeed using Jetstream with Inertia, and yes, I expect Laravel to have updated my form after the submission. However, I don't use `resetOnSuccess: true` in my form, I'm not even sure how to use it. However, in [this page](https://jetstream.laravel.com/1.x/stacks/inertia.html#form-validation-helpers), they're talking about this [npm package](https://github.com/laravel/jetstream-js), I don't know if I should install it.

At the end of the day I could also redirect the user to some other place just to be sure that everything would work properly. I don't know what to do here..

**Comment by claudiodekker** [comment]
Hi, you don't have to install that package, as Jetstream already installed that for you.

I've done some digging into what could be happening, and discovered that Jetstream's Inertia 'form' helper [resets the form to the initial values automatically](https://github.com/laravel/jetstream-js/blob/master/src/InertiaForm.js#L64) (using that `resetOnSuccess` value @reinink mentioned earlier). This explains why you're seeing the form being reset to the 'old' values (which are really being considered the 'initial' values)

What you'll probably want to do instead, is to chain a `.then()` on top of `this.landing.post` request (the Jetstream form helper doesn't support the new event system callbacks yet), and to 'reset' the values from the props at that point:

```js
this.landing.post('/some/url').then(() => {
  if(this.landing.recentlySuccessful) {
    this.landing.setInitialValues({
      ... this.landing.data(),
      ... this.$page.landing
    })
    this.landing.reset()
  }
});

I haven't tested this, but what this basically does it:
1. Check if the request was successful
2. Get the 'initial' form values, and combine them with the $page.landing properties
3. Write back the 'new' initial values
4. 'reset' the form, setting the new valeus

I hope this works & helps. I didn't actually test it, but I believe it (or something similar) should work.
In case you want to look into this yourself, here's the logic for the form itself: https://github.com/laravel/jetstream-js/blob/f3ad70784c3deafa9d669c0b86130ebf13117f06/src/InertiaForm.js

**Comment by Drillan767** [comment]
Hello so, bad news and good news:

Bad news is, I tried your code (thanks for your time and the quick answer, by the way!), and it didn't work. I was at the verge of just setting a `window.location.reload()` so that I'm sure to have my data... until I wondered if `this.$page.landing` was updating or not. Well, it was.

So, I did the following, I just don't know how ugly and hacky it works... but it works.

```
this.landing.post('/some/url')
    .then(() => {
        this.landing = this.$inertia.form({
            ...this.$page.landing
        })
    });
```
